### PR TITLE
 Add BLE in Neuromore

### PR DESCRIPTION
**Pull request by Sarthak031**
Adding functionality to use inbuilt Bluetooth adaptor in Neuromore to visualize data over BLE.

### CODE REVIEW

**Comment by Sarthak031** [comment]
Please Review my code if it is not acceptable please give me suggestions for this so that I can improve it.

**Comment by dioptre** [comment]
Mate this PR has a lot more code in it than just BLE, and you've turned application wide variables on and off. Please just commit what is necessary.

**Comment by Sarthak031** [comment]
@dioptre 
Hi Andrew,
Please refer to my [Add BLE Functionality](https://github.com/neuromore/studio/pull/100) PR. In that PR There is not only the BLE functionality added but there are a lot more things just like-

- Brainalive Device added using the BLE functionality.
- Channel
- 

> Mate this PR has a lot more code in it than just BLE, and you've turned application wide variables on and off. Please just commit what is necessary.

File: build/vs/BrainaliveDevices.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
+/****************************************************************************
```
Comment (dioptre) [inline build/vs/BrainaliveDevices.cpp:1]: This file needs to live in Engine/Devices

File: src/Engine/Devices/Generic/GenericDeviceNodes.h [hunk 2]
```diff
@@ -43,7 +43,7 @@ class HeartRateNode : public DeviceInputNode
 		Core::Color GetColor() const override						{ return Core::RGBA(102,45,145); }
 		uint32 GetType() const override								{ return TYPE_ID; }
 		const char* GetTypeUuid() const override final				{ return Uuid(); }
-		const char* GetReadableType() const override				{ return "Heartrate"; }
+		const char* GetReadableType() const override				{ return "HeartRate"; }
```
Comment (dioptre) [inline src/Engine/Devices/Generic/GenericDeviceNodes.h:46]: Why is this changed?

File: src/Engine/EEGElectrodes.cpp [hunk 3]
```diff
@@ -76,7 +76,14 @@ void EEGElectrodes::Init()
     mElectrodes.Add( Electrode("O1",  -108.000,  0.000) );
     mElectrodes.Add( Electrode("O2",  -72.000,  0.000) );
     mElectrodes.Add( Electrode("Oz",  -90.000,  0.000) );
-
+	mElectrodes.Add(Electrode("Ch1", 90.000, 0.000));
```
Comment (dioptre) [inline src/Engine/EEGElectrodes.cpp:79]: Not sure this is the best way to do this?

File: src/Engine/User.cpp [hunk 4]
```diff
@@ -58,7 +59,8 @@ Core::String User::CreateDisplayableName() const
 
 
 Core::String User::CreateFullName() const
-{
+{
```
Comment (dioptre) [inline src/Engine/User.cpp:62]: These changes shouldn't be committed

File: src/Engine/User.h [hunk 5]
```diff
@@ -106,7 +106,14 @@ class ENGINE_API User
 		// rule permissions
 		const Rule* FindRule(const char* ruleName) const;
 		bool CreateAllowed(const char* ruleName)  const { if (const Rule* r = FindRule(ruleName)) return r->GetCreud().Create(); else return false; }
-		bool ReadAllowed(const char* ruleName)    const { if (const Rule* r = FindRule(ruleName)) return r->GetCreud().Read(); else return false; }
+		bool ReadAllowed(const char* ruleName)
```
Comment (dioptre) [inline src/Engine/User.h:109]: Why did you change the User?

File: src/Studio/AppManager.h [hunk 6]
```diff
@@ -90,10 +90,6 @@ class AppManager : public QObject
 		const char* GetLicenseUrl() const								{ return "https://assets.eego-perform.com/license/license.txt"; }
 		const char* GetCloudTermsUrl() const							{ return "https://assets.eego-perform.com/license/terms-and-conditions.txt"; }
 		const char* GetPrivacyPolicyUrl() const							{ return "https://assets.eego-perform.com/license/privacy-policy.txt"; }
-
```
Comment (dioptre) [inline src/Studio/AppManager.h:93]: Why?

File: src/Studio/Plugins/Devices/BciDeviceWidget.cpp [hunk 7]
```diff
@@ -27,12 +27,6 @@
 
 using namespace Core;
 
-#if defined(NEUROMORE_BRANDING_ANT) && defined(PRODUCTION_BUILD)
```
Comment (dioptre) [inline src/Studio/Plugins/Devices/BciDeviceWidget.cpp:30]: Why?

File: src/Studio/Plugins/Devices/BciDeviceWidget.cpp [hunk 8]
```diff
@@ -135,12 +129,10 @@ void BciDeviceWidget::AddSensorItems(QTreeWidgetItem* parent)
 		// add signal quality subitem
 		item = new QTreeWidgetItem(sensorItem);
 		item->setText(0, "Signal Quality");
-		item->setHidden(NEUROMORE_MINIMAL_DEVICE_STATS);
```
Comment (dioptre) [inline src/Studio/Plugins/Devices/BciDeviceWidget.cpp:138]: Theres so much you've changed that I'm dubious about. Please resubmit the PR again.

**Review by dioptre — CHANGES_REQUESTED** [review]
Please remove all irrelevant code changes and re-submit.

File: build/vs/BrainaliveDevices.cpp (hunk 1 above)
Comment (Sarthak031) [reply build/vs/BrainaliveDevices.cpp:1]: @dioptre as I already mentioned in my previous comment please refer to [Add BLE Functionality (#100)](https://github.com/neuromore/studio/pull/100) In this PR I already move these files into the Engine/Devices. but now you closed (#100) PR so I resubmit the PR.

File: src/Studio/Plugins/Devices/BciDeviceWidget.cpp (hunk 8 above)
Comment (Sarthak031) [reply src/Studio/Plugins/Devices/BciDeviceWidget.cpp:138]: @dioptre 
Please check my new PR [New_PR(#102)](https://github.com/neuromore/studio/pull/102). I have changed a few things according to your issues in this PR.
